modules/xfem/src/userobjects/GeometricCutUserObject.C

```diff
@@ -34,14 +37,25 @@ validParams<GeometricCutUserObject>()
 }
 
 GeometricCutUserObject::GeometricCutUserObject(const InputParameters & parameters)
-  : CrackFrontPointsProvider(parameters)
+  : CrackFrontPointsProvider(parameters), _heal_mesh(getParam<bool>("heal_mesh"))
 {
-  FEProblemBase * fe_problem = dynamic_cast<FEProblemBase *>(&_subproblem);
-  if (fe_problem == NULL)
-    mooseError("Problem casting _subproblem to FEProblemBase in XFEMMaterialStateMarkerBase");
-  _xfem = MooseSharedNamespace::dynamic_pointer_cast<XFEM>(fe_problem->getXFEM());
+  _xfem = MooseSharedNamespace::dynamic_pointer_cast<XFEM>(_fe_problem.getXFEM());
```

Comment:
Don't use `MooseSharedNamespace` anymore, it's deprecated. Just use the C++11 smart pointer names directly.

Reply:
I will make the changes to other files too.